src/worker/tasks/processTx.ts

```diff
@@ -152,469 +108,279 @@ export const processTx = async () => {
-                      extension,
-                      queuedAt,
-                    };
-                  }
-                },
-              ),
-            );
-
-            // - After sending transactions, update database for each transaction
-            await Promise.all(
-              txStatuses.map(async (tx) => {
-                switch (tx.status) {
-                  case TransactionStatusEnum.Submitted:
-                    await updateTx({
-                      pgtx,
-                      queueId: tx.queueId,
-                      data: {
-                        sentAt: tx.sentAt,
-                        status: TransactionStatusEnum.Submitted,
-                        transactionHash: tx.transactionHash,
-                        res: tx.res,
-                        sentAtBlockNumber: await provider.getBlockNumber(),
-                      },
-                    });
-                    reportUsageForQueueIds.push({
-                      input: {
-                        fromAddress: tx.res?.from,
-                        toAddress: tx.res?.to,
-                        value: tx.res?.value?.toString(),
-                        chainId: tx.res?.chainId?.toString(),
-                        transactionHash: tx.transactionHash,
-                        functionName: tx.functionName || undefined,
-                        extension: tx.extension || undefined,
-                        provider: provider.connection.url || undefined,
-                        transactionValue: tx.res?.value?.toString(),
-                        msSinceQueue: new Date().getTime() - tx.queuedAt!,
-                      },
-                      action: UsageEventTxActionEnum.SendTx,
-                    });
-                    break;
-                  case TransactionStatusEnum.Errored:
-                    await updateTx({
-                      pgtx,
-                      queueId: tx.queueId,
-                      data: {
-                        status: TransactionStatusEnum.Errored,
-                        errorMessage: tx.errorMessage,
-                      },
-                    });
-                    reportUsageForQueueIds.push({
-                      input: {
-                        fromAddress: tx.txRequest.from,
-                        toAddress: tx.txRequest.to,
-                        value: tx.txRequest.value?.toString(),
-                        chainId: tx.txRequest.chainId?.toString(),
-                        functionName: tx.functionName || undefined,
-                        extension: tx.extension || undefined,
-                        provider: provider.connection.url || undefined,
-                        msSinceQueue: new Date().getTime() - tx.queuedAt!,
-                      },
-                      action: UsageEventTxActionEnum.NotSendTx,
-                    });
-                    break;
-                }
-              }),
-            );
-          } catch (err: any) {
```

Comment:
Removing this `catch` block. I don't think we should error out txs if this code block throws because it's likely unrelated issues (RPC, wallet).

Reply:
this try/catch helps in catching these:

```
QUEUED:INCREASE_ALLOWANCE:ENGINE:ENGINE_FAILURE:[Worker] [Error] Failed to process batch of transactions for wallet - Error: timeout (requestBody="[{\"method\":\"eth_blockNumber\",\"params\":[],\"id\":5521,\"jsonrpc\":\"2.0\"}]", requestMethod="POST", timeout=120000, url="https://polygon.rpc.thirdweb.com/fc01ea4fbd46234e23098c7daa7b6ef2", code=TIMEOUT, version=web/5.7.1)
```

this line, i believe, `sentAtBlockNumber: await provider.getBlockNumber(),` is getting caught there.

similarly, other RPC calls to get tx details etc gets caught here. maybe we need to update this logic

Maybe we can add a try-catch inside the match and handle the updatesToDB when transasction was processed but couldn't get data via RPC more gracefully, instead of just marking them as errored.